addons/website_slides/models/slide_channel.py

```diff
@@ -45,8 +45,18 @@ def _compute_completion(self):
             slide_done = mapped_data.get(record.channel_id.id, dict()).get(record.partner_id.id, 0)
             slide_total = channel_data.get(record.channel_id.id) or 1
             record.completion = math.ceil(100.0 * slide_done / slide_total)
-            if record.completion >= 100:
-                record.update({'completed': True})
+
+    @api.multi
+    def _write(self, vals):
+        if 'completion' in vals and vals['completion'] >= 100:
+            vals.update({'completed': True})
+            partner_has_completed = {}
+            for record in self:
+                partner_has_completed.update({record.partner_id.id: record.channel_id})
```

Comment:
dict comprehension : `partner_has_completed = {(record.partner_id.id: record.channel_id) for record in self}`
And avoid `record`, as we know this is a `channel_partner` ;)